playback/src/audio_backend/alsa.rs

```diff
@@ -1,14 +1,81 @@
 use super::{Open, Sink};
-use alsa::{Access, Format, Mode, Stream, PCM};
+use alsa::device_name::HintIter;
+use alsa::pcm::{Access, Format, HwParams, PCM};
+use alsa::{Direction, Error, ValueOr};
+use std::env;
+use std::ffi::CString;
 use std::io;
+use std::process::exit;
 
 pub struct AlsaSink(Option<PCM>, String);
 
+fn list_outputs() {
+    for t in &["pcm", "ctl", "rawmidi", "timer", "seq", "hwdep"] {
```

Comment:
Since we are exclusively working with PCM devices, do we really need to show the other interfaces? `rawmidi`, `seq` and `timer` are particularly uninteresting.

Reply:
Good point - this is stray code from debugging times. I don't think this even works now. Will clean it up.

Reply:
was going to say, didn't look permanent :)